sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerProviderBuilder.java

```diff
@@ -147,14 +151,26 @@ public SdkTracerProviderBuilder addSpanProcessor(SpanProcessor spanProcessor) {
     return this;
   }
 
+  public SdkTracerProviderBuilder setTracerConfigProvider(
```

Comment:
Not blocking this PR, but I think we'll need to add `addTracerConfigProvider` at some point.

I'm trying to model how the following case can be supported (which I consider to be very common):
- distro wants to enable certain scopes and disable others by default
- end user comes and wants to enable/disable a few more scopes, but does not want to override the whole distro filter

I assume we'd introduce `addTracerConfigProvider`.
Another option that autoconfiguration would provide a customizer that would build composite provider. Or distros would be forced to provide some config API to let users customize the config provider.

Leaving autoconfiguration aside, with vanilla `addTracerConfigProvider` we'd need to come up with conflict resolution approach (one component enables the scope, another disables the same one).

Reply:
That's a good point. We could say that if the SDK interprets a response of `null` from `Function<InstrumentationScopeInfo, TracerConfig>` as indicating that it has no preference, and to use the SDK system default. And when a user calls `addTracerConfigProvider`, we could compose a higher order function which calls previously added functions in order, and returns the first non-null value. 

> Leaving autoconfiguration aside, with vanilla addTracerConfigProvider we'd need to come up with conflict resolution approach (one component enables the scope, another disables the same one).

First config provider with an opinion wins? Then it becomes a distro problem to make sure that it registers its default scopes in the right order with respect to end user customizations.